### PR TITLE
Fix rustc-args passing issue in bootstrap

### LINKED ISSUE #113178: bootstrap: `rustc-args` is only passed to compiletest, not unit tests
https://github.com/rust-lang/rust/blob/5122e88b35dbf4e959d6ca2cfca4c5ed0a105203/src/bootstrap/test.rs#L1586

cc https://rust-lang.zulipchat.com/#narrow/stream/219381-t-libs/topic/Running.20tests.20in.20debug.20mode.20with.20debug.20symbols.3F/near/361025470

We should either pass this flag consistently or rip it out entirely; having it work only sometimes makes it an attractive nuisance.

### Meta

HEAD is 330727467b8fdf2c43b95095a0efae7012c4f83b

### PR DESCRIPTION
**Pull request by chenyukang**
Fixes #113178, r? @jyn514

### CODE REVIEW

**Comment by lqd** [comment]
https://rust-lang.zulipchat.com/#narrow/stream/122651-general/topic/vacation/near/376793983

r? bootstrap

File: src/bootstrap/builder.rs [hunk 1]
```diff
@@ -2043,6 +2043,13 @@ impl<'a> Builder<'a> {
             rustflags.arg("-Zinline-mir");
         }
 
+        // set rustc args passed from command line
+        let rustc_args =
+            self.config.cmd.rustc_args().iter().map(|s| s.to_string()).collect::<Vec<_>>();
+        if !rustc_args.is_empty() {
+            cargo.env("RUSTFLAGS", &rustc_args.join(" "));
+        }
```
Comment (RalfJung) [inline src/bootstrap/builder.rs:2051]: I don't understand this patch, it doesn't make sense to me. This is setting the RUSTFLAGS env var, but when this `Cargo` instance is later turned into a `Command`, the RUSTFLAGS env var is anyway overwritten again:

https://github.com/rust-lang/rust/blob/c350d3c5dd00ed0727ffe72fc33637d8c7537733/src/bootstrap/src/core/builder.rs#L2414-L2418

Comment (RalfJung) [reply src/bootstrap/builder.rs:2051]: Shouldn't this be added to `rustflags` instead?

Comment (RalfJung) [reply src/bootstrap/builder.rs:2051]: Also the --help text says this should only be passed when running tests; I don't see that reflected in the code.

Comment (onur-ozkan) [reply src/bootstrap/builder.rs:2051]: Yeah, this isn't correct and fixed in https://github.com/rust-lang/rust/pull/123489.

**Pull request merged**